ext/intl/formatter/formatter_format.c

```diff
@@ -76,7 +98,13 @@ PHP_FUNCTION( numfmt_format )
 
 		case FORMAT_TYPE_INT64:
 		{
-			int64_t value = (Z_TYPE_P(number) == IS_DOUBLE)?(int64_t)Z_DVAL_P(number):Z_LVAL_P(number);
+			int64_t value;
+			if (Z_TYPE_P(number) == IS_DOUBLE) {
+				value = (int64_t)Z_DVAL_P(number);
```

Comment:
AFAIK, this conversion is undefined behavior if the double value is outside the range of `int64_t`. However, we already do that prior to this patch, so it may be okay for all supported platforms.

Reply:
As it turns out, your comment was prescient: the s390 architecture got `+9223372036854775808` instead of `-9223372036854775808` as a result of that cast.  So I've tweaked the test to allow any value for this case.